src/csharp/Grpc.Core/ForwardedTypes.cs

```diff
@@ -18,21 +18,28 @@
 
 using System.Runtime.CompilerServices;
 using Grpc.Core;
-using Grpc.Core.Logging;
+using Grpc.Core.Internal;
 using Grpc.Core.Utils;
 
 // API types that used to be in Grpc.Core package, but were moved to Grpc.Core.Api
 // https://docs.microsoft.com/en-us/dotnet/framework/app-domains/type-forwarding-in-the-common-language-runtime
 
-// TODO(jtattermusch): move types needed for implementing a client
-
 [assembly:TypeForwardedToAttribute(typeof(GrpcPreconditions))]
+[assembly:TypeForwardedToAttribute(typeof(AsyncClientStreamingCall<,>))]
+[assembly:TypeForwardedToAttribute(typeof(AsyncDuplexStreamingCall<,>))]
+[assembly:TypeForwardedToAttribute(typeof(AsyncServerStreamingCall<>))]
+[assembly:TypeForwardedToAttribute(typeof(AsyncUnaryCall<>))]
 [assembly:TypeForwardedToAttribute(typeof(AuthContext))]
```

Comment:
What about AuthInterceptorContext in AsyncAuthInterceptor.cs?

Reply:
Good catch. Added.